lesson16/js/main.js

```diff
@@ -0,0 +1,169 @@
+	loadingImg.src = 'images/loading-circle.gif';
+	loadingImg.classList.add('el_loadingImg');
+	newsContentArea.appendChild(createLoadingBox).appendChild(loadingImg);
+}
+
+function hideLoadingImg() {
+	const loading = document.getElementById('js-loading');
+	loading.remove();
+}
+
+function createTabContent(){
+		const newsContentArea = document.createElement('div');
+		newsContentArea.classList.add('news-area');
+		newsContentArea.setAttribute('role','tabpanel');
+		categoryTab.parentNode.insertBefore(newsContentArea,categoryTab.nextSibling);
+}
+
+function renderCategoryTab(data){
+	const fragment = document.createDocumentFragment();
+
+	data.forEach((category,i) => {
+		const categoryTab = document.createElement('li');
+		categoryTab.classList.add('tab_list_item');
+		categoryTab.setAttribute('role','presentation');
+
+		const categoryButton = document.createElement('button');
+		categoryButton.textContent = category.name;
+		categoryButton.setAttribute('class', 'tab-button js-tabButton');
+		categoryButton.setAttribute('aria-selected', 'false');
+		categoryButton.setAttribute('role', 'tab');
+		categoryButton.setAttribute('tabindex', '-1');
+		fragment.appendChild(categoryTab).appendChild(categoryButton);
+	});
+
+	const categoryTabLists = document.querySelector('.tab-list');
+	categoryTabLists.appendChild(fragment);
+}
+
+function categoryTabInitial(){
+	const firstTab = document.querySelectorAll('.tab-button')[0];
+	firstTab.setAttribute('aria-selected','true');
+	firstTab.setAttribute('tabindex','0');
+}
+
+function renderNewsArticle(data){
+	const newsContentArea = document.querySelector('.news-area');
+	const newsLists = document.createElement('ul');
+	newsLists.classList.add('news-list');
+	const fragment = document.createDocumentFragment();
+
+	const displayedResults = 5;
+	for (let i = 0; i < displayedResults; ++i){
+		const li = document.createElement('li');
+		li.classList.add('news-item');
+
+		const anchor = document.createElement('a');
+		anchor.classList.add('news-link');
+		anchor.href = data[i].link;
+		anchor.textContent = data[i].title;
+		fragment.appendChild(li).appendChild(anchor);
+	}
+
+	newsContentArea.appendChild(newsLists).appendChild(fragment);
+};
+
+function resetArticle(){
+	const newsLists= document.querySelector('.news-list');
+	newsLists.remove();
+}
+
+function displayErrorMassage(error){
+	const getNewsArea = document.querySelectorAll('.news-area');
+	for(let i = 0; i < getNewsArea.length; i++){
+		const createTextBox = document.createElement('p');
+		createTextBox.textContent = error;
+		getNewsArea[i].appendChild(createTextBox);
+	}
+}
+
+async function fetchNewsData() {
+	const NEWS_DATA_URL = 'https://api.json-generator.com/templates/2PqhEvPcqUZW/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl';
+	try {
+		const response = await fetch(NEWS_DATA_URL);
+		if(response.ok){
+			const json = await response.json();
+			return json;
+		} else{
+			throw new Error('Network Error');
+		}
```

Comment:
IMO
On Discord, Moriken-sensei said.(Thread:task14)
> サーバーからのレスポンスエラー50xはエラーをスローしてはいけないです。サーバーは意図があって返しているからです。現場の指針によりますが、それなりに対応して画面に表示させるのがよいです
ここの課題では画面を分ける必要はないですが、コンソールエラーぐらいにしておいてもらいたいです 
エラーをスローするということは画面がクラッシュすることを意味していて、開発段階では開発者に気づかせる意味でやりますが、本番を想定している課題画面ではクラッシュさせないようにしてください

Also, what do you think about returning the message in this way because it is difficult to understand the details if the message is returned as a string?
```suggestion
    if (!response.ok) {
      console.error(`${response.status}:${response.statusText}`);
    }
    return await response.json();
```

Reply:
Because the method you taught me was so simple.
We have modified it here.

If I just display logs, the screen goes blank.
The text display is also retained.

[e20f968](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/e20f968a36fa3cac8247d8e3b04cd75747d1a0bf)

Reply:
[e20f968](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/e20f968a36fa3cac8247d8e3b04cd75747d1a0bf)
In this modification, you are handling exceptions, right?

> サーバーからのレスポンスエラー50xはエラーをスローしてはいけないです。

May I ask you to check again?

Reply:
It turns out that throwing an error here limits the handling of the error or the amount of code is huge in error handling!
his has been fixed.

[e20f968](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/e20f968a36fa3cac8247d8e3b04cd75747d1a0bf)